### PR TITLE
Add a class that provides balance check, transfer functions for one KeyPair

### PR DESCRIPTION
**Pull request by MichaelKim20**
Many functions in SKD must be called sequentially for boa transfer. This can be difficult to use the SDK.
I wanted to make the transfer and balance check possible in one function call.
And I'm planning to add freeze and release function next time.

Related to #204

### CODE REVIEW

File: src/modules/net/BOAClient.ts [hunk 1]
```diff
@@ -282,13 +283,37 @@ export class BOAClient {
...
         return Request.get(url.toString()).then((response: AxiosResponse) => {
             return JSBI.BigInt(response.data);
         });
     }
 
+    public getStoaStatus(): Promise<boolean> {
+        let url = uri(this.server_url).directory("/block_height");
+
+        return Request.get(url.toString())
+            .then((response: AxiosResponse) => {
```
[the remark this answers is not in the document]
Comment (MichaelKim20) [reply src/modules/net/BOAClient.ts:297]: The functionality is already built into Axios.
`then()` Internal is called only when there are 200 units.
`catch()` is called when it is a value other than 200.

File: src/modules/net/BOAClient.ts [hunk 2]
```diff
@@ -473,6 +498,26 @@ export class BOAClient {
...
+     * @returns Promise that resolves or
+     * rejects with response from the Stoa
+     */
+    public getBalance(address: PublicKey): Promise<Balance> {
+        return new Promise<Balance>((resolve, reject) => {
+            let url = uri(this.server_url).directory("wallet/balance").filename(address.toString());
+
+            Request.get(url.toString())
+                .then((response: AxiosResponse) => {
+                    resolve(Balance.reviver("", response.data));
```
[the remark this answers is not in the document]
Comment (MichaelKim20) [reply src/modules/net/BOAClient.ts:513]: The functionality is already built into Axios.
`then()` Internal is called only when there are 200 units.
`catch()` is called when it is a value other than 200.

File: src/modules/wallet/Wallet.ts [hunk 3]
```diff
@@ -0,0 +1,435 @@
...
+            return { code: WalletResultCode.FailedAccessToAgora, message: e.message };
+        }
+
+        // Stoa Access Test
+        try {
+            if (!(await this.client.getStoaStatus()))
+                return { code: WalletResultCode.FailedAccessToStoa, message: "Failed access to Stoa." };
+        } catch (e) {
+            return { code: WalletResultCode.FailedAccessToStoa, message: e.message };
+        }
```
[the remark this answers is not in the document]
Comment (MichaelKim20) [reply src/modules/wallet/Wallet.ts]: That's a good point. So I found a good way and modified it.

**Comment by MichaelKim20** [comment]
I performed the test using the code below in the actual network and confirmed that it was running well.
``` TypeScript
let res: sdk.IWalletResult;
try {
    let key_count = this.config.process.key_count;
    let sources: Array<number> = [];
    let source = Math.floor(Math.random() * key_count);
    while (sources.find((value) => value == source) !== undefined)
        source = Math.floor(Math.random() * key_count);
    sources.push(source);
    let source_key_pair = WK.keys(source);

    let destination = Math.floor(Math.random() * key_count);
    while (source === destination) destination = Math.floor(Math.random() * key_count);
    let destination_key_pair = WK.keys(destination);

    let wallet = new sdk.Wallet(source_key_pair, {
        agoraEndpoint: this.config.server.agora_endpoint.toString(),
        stoaEndpoint: this.config.server.stoa_endpoint.toString(),
        fee: sdk.WalletFeeOption.Medium,
    });
    let bal_res = await wallet.getBalance();
    if (bal_res.code !== sdk.WalletResultCode.Success || bal_res.balance === undefined) {
        return resolve({
            status: false,
            error: bal_res.message,
        });
    }
    let range = sdk.JSBI.BigInt(Math.floor(Math.random() * 50) + 20);
    let send_amount = sdk.JSBI.divide(
        sdk.JSBI.multiply(bal_res.balance.spendable, range),
        sdk.JSBI.BigInt(100)
    );
    res = await wallet.transfer([
        {
            address: destination_key_pair.address,
            amount: send_amount,
        },
    ]);
    if (res.code === sdk.WalletResultCode.Success) {
        let h = sdk.hashFull(res.tx).toString();
        logger.info(`TX_HASH (send / ${height.toString()}) : ${h}`);
        return resolve({
            status: true,
            data: h,
        });
    } else {
        return resolve({
            status: false,
            error: res.message,
        });
    }
} catch (e) {
    return resolve({
        status: false,
        error: e,
    });
}
```

**Comment by MichaelKim20** [comment]
I performed the test using the code below in the actual network and confirmed that it was running well.
``` TypeScrypt
let res: sdk.IWalletResult;
try {
    let wallet = new sdk.Wallet(WK._keys[0], {
        agoraEndpoint: this.config.server.agora_endpoint.toString(),
        stoaEndpoint: this.config.server.stoa_endpoint.toString(),
        fee: sdk.WalletFeeOption.Medium,
    });
    res = await wallet.cancelWithHash(this.tx_hash, (addresses: sdk.PublicKey[]) => {
        return WK._keys.filter(
            (w) => addresses.find((m) => Buffer.compare(w.address.data, m.data) === 0) !== undefined
        );
    });
    if (
        res.code === sdk.WalletResultCode.Success &&
        res.tx !== undefined &&
        res.original_tx !== undefined
    ) {
        let h = sdk.hashFull(res.tx).toString();
        logger.info(`TX_HASH (send / ${height.toString()}) : ${h}`);
        return resolve({
            status: true,
            cancel_tx: res.tx,
            cancel_tx_hash: sdk.hashFull(res.tx).toString(),
            cancel_tx_size: res.tx.getNumberOfBytes(),
            original_tx: res.original_tx,
            original_tx_hash: sdk.hashFull(res.original_tx).toString(),
            original_tx_size: res.original_tx.getNumberOfBytes(),
        });
    } else {
        return resolve({
            status: false,
            error: res.message,
        });
    }
} catch (e) {
    return resolve({
        status: false,
        error: e,
    });
}
```

File: src/modules/wallet/Wallet.ts [hunk 4]
```diff
@@ -0,0 +1,521 @@
...
+        let tx: Transaction;
+        try {
+            tx = await this.client.getPendingTransaction(tx_hash);
+        } catch (e) {
+            return { code: WalletResultCode.FailedRequestPendingTransaction, message: e.message };
+        }
+        return this.cancel(tx, key_finder);
+    }
+
+    public static createKeyPair(): KeyPair {
```
Comment (TrustHenry) [inline src/modules/wallet/Wallet.ts:514]: Where is it used?

File: src/modules/wallet/Wallet.ts [hunk 5]
```diff
@@ -0,0 +1,526 @@
...
+        return KeyPair.random();
+    }
+
+    public clearUTXOCache() {
```
Comment (TrustHenry) [inline src/modules/wallet/Wallet.ts:523]: It is recommended that add a document.

**Pull request merged**